lib/check-dependencies.js

```diff
@@ -0,0 +1,149 @@
+import { fs, logger } from 'appium-support';
+import { getDevices } from 'node-simctl';
+import { asyncify } from 'asyncbox';
+import _ from 'lodash';
+import { exec } from 'teen_process';
+import path from 'path';
+import { EOL } from 'os';
+import { fileCompare, WEBDRIVERAGENT_PROJECT } from './utils';
+import { BOOTSTRAP_PATH} from './constants';
+
+const log = logger.getLogger('WebDriverAgent');
+const execLogger = {
+  // logger that gets rid of empty lines
+  logNonEmptyLines (data, fn) {
+    data = Buffer.isBuffer(data) ? data.toString() : data;
+    for (const line of data.split(EOL)) {
+      if (line) {
+        fn(line);
+      }
+    }
+  },
+  debug (data) {
+    this.logNonEmptyLines(data, log.debug.bind(log));
+  },
+  error (data) {
+    this.logNonEmptyLines(data, log.error.bind(log));
+  },
+};
+
+const IOS = 'iOS';
+const TVOS = 'tvOS';
+
+const CARTHAGE_CMD = 'carthage';
+const CARTFILE = 'Cartfile.resolved';
+const CARTHAGE_ROOT = 'Carthage';
+
+let buildDirPath;
+
+async function hasTvOSSims () {
+  const devices = _.flatten(Object.values(await getDevices(null, TVOS)));
+  return !_.isEmpty(devices);
+}
+
+function getCartfileLocations () {
+  const cartfile = path.resolve(BOOTSTRAP_PATH, CARTFILE);
+  const installedCartfile = path.resolve(BOOTSTRAP_PATH, CARTHAGE_ROOT, CARTFILE);
+
+  return {
+    cartfile,
+    installedCartfile,
+  };
+}
+
+async function needsUpdate (cartfile, installedCartfile) {
+  return !await fileCompare(cartfile, installedCartfile);
+}
+
+async function fetchDependencies (useSsl = false) {
+  log.info('Fetching dependencies');
+  if (!await fs.which(CARTHAGE_CMD)) {
+    log.errorAndThrow('Please make sure that you have Carthage installed (https://github.com/Carthage/Carthage)');
```

Comment:
I would also add a recommendation to check PATH, because some people have it installed, but not visible to Appium